File: Edward Poot/src/QL/ast/expr/binaryoperatorexpr/add.go
package binaryoperatorexpr

import ("ql/ast/visit"
"ql/ast/expr")

type Add struct {
	Lhs, Rhs expr.Expr
}

func (a Add) GetLhs() expr.Expr {
	return a.Lhs
}

func (a Add) GetRhs() expr.Expr {
	return a.Rhs
}

func (a Add) Eval() interface{} {
	return a.GetLhs().Eval().(int) + a.GetRhs().Eval().(int)
}

func (a Add) Accept(v visit.Visitor) interface{} {
	return v.Visit(a)
}

File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/math/Add.java
package org.uva.sea.ql.ast.expr.math;

import org.uva.sea.ql.ast.expr.BinaryExpr;
import org.uva.sea.ql.ast.expr.Expr;
import org.uva.sea.ql.ast.form.TypeMap;
import org.uva.sea.ql.ast.form.ValueMap;
import org.uva.sea.ql.type.*;
import org.uva.sea.ql.value.Value;
import org.uva.sea.ql.visit.Visitor;

public class Add extends BinaryExpr {
	
	public Add(Expr lhs, Expr rhs, int startLine) {
		super(lhs, rhs, startLine);
	}
	
	@Override
	public Value eval(ValueMap valueMap) {
		return lhs.eval(valueMap).add(rhs.eval(valueMap)); 
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (+)";
	}

	@Override
	public Type getType(TypeMap context) {
		return new NumericalType();
	}
}
